dagshub/streaming/dataclasses.py

```diff
@@ -24,15 +26,35 @@ class DagshubPath:
         relative_path (Optional[Path]): Path relative to the root of the encapsulating FileSystem.
                                         If None, path is outside the FS
         original_path (Path): Original path as it was accessed by the user
+        is_binary_path_requested (bool): For functions like scandir and listdir that have
+            different behaviour whether user requested a string or a binary path
     """
 
-    # TODO: this couples this class hard to the fs, need to decouple later
-    fs: "DagsHubFilesystem"  # Actual type is DagsHubFilesystem, but imports are wonky
-    absolute_path: Optional[Path]
-    relative_path: Optional[Path]
-    original_path: Optional[Path]
+    def __init__(self, fs: "DagsHubFilesystem", file_path: Union[str, bytes, PathLike, "DagshubPath"]):
+        self.fs = fs
+        self.is_binary_path_requested = isinstance(file_path, bytes)
+        self.absolute_path, self.relative_path, self.original_path = self.parse_path(file_path)
 
-    def __post_init__(self):
+    def parse_path(self, file_path: Union[str, bytes, PathLike, "DagshubPath"]) -> Tuple[Path, Optional[Path], Path]:
+        if isinstance(file_path, DagshubPath):
+            self.is_binary_path_requested = file_path.is_binary_path_requested
+            if file_path.fs != self.fs:
+                relativized = DagshubPath(self.fs, file_path.absolute_path)
+                return relativized.absolute_path, relativized.relative_path, relativized.original_path
+            return file_path.absolute_path, file_path.relative_path, file_path.original_path
+        if isinstance(file_path, bytes):
+            file_path = os.fsdecode(file_path)
+        orig_path = Path(file_path)
```

Comment:
I want to have the og requested path as a Path object because it's more convenient to use.
Having the original path be both a Path object and an additional `PathType` would be confusing w.r.t. "which one should I be using", so I kept just the single thing I care about there, which is whether the requested path was bytes or not

Reply:
I checked the usages, and it's probably alright to actually just carry over the original path, because I'm not working with it that much

Reply:
Changed it to carry over as-is